monopod/pkg/commands/scaffold.go

```diff
@@ -136,72 +181,78 @@ func (o *scaffoldOptions) validateOptions() error {
+	source := fmt.Sprintf("./images/%s", o.ModuleName)
+	targetRepo := fmt.Sprintf(` "${var.target_repository}/%s"`, o.ModuleName)
+	// use a Token here because hclwrite escapes things like $ in strings
+	// even using literals or double escaping doesn't help. See this go.cty issue:
+	// https://github.com/zclconf/go-cty/issues/163#issuecomment-1754073585
+	toks := hclwrite.Tokens{
+		&hclwrite.Token{
+			Type:         hclsyntax.TokenStringLit,
+			Bytes:        []byte(targetRepo),
+			SpacesBefore: 0,
+		},
+	}
 
-	// Define the target repository
-	targetRepository := "${var.target_repository}"
+	newModule := hclwrite.NewBlock("module", []string{o.ModuleName})
+	newModule.Body().SetAttributeValue("source", cty.StringVal(source))
+	newModule.Body().SetAttributeRaw("target_repository", toks)
+
+	f, diags := hclwrite.ParseConfig(o.MainTf, "main.tf", hcl.Pos{
+		Line:   0,
+		Column: 0,
+		Byte:   0,
+	})
+	if diags != nil {
+		return fmt.Errorf("%v", diags)
+	}
 
-	file, err := os.OpenFile("main.tf", os.O_RDWR, os.ModePerm)
-	if err != nil {
-		fmt.Printf("Error opening file: %v\n", err)
-		os.Exit(1)
-	}
-	defer file.Close()
-
-	// Create a slice to hold the lines of the updated content
-	var updatedContent []string
-
-	// Track whether the module block has been inserted
-	moduleInserted := false
-
-	scanner := bufio.NewScanner(file)
-	for scanner.Scan() {
-		line := scanner.Text()
-
-		// Check if this is where the module block should be inserted
-		if !moduleInserted && strings.HasPrefix(line, "module ") {
-			l := strings.TrimPrefix(strings.ReplaceAll(strings.TrimSuffix(line, " {"), "\"", ""), "module ")
-			if strings.Compare(o.PackageName, l) == 0 {
-				return fmt.Errorf("module %s block already exists in main.tf", o.PackageName)
-			}
-			if strings.Compare(o.PackageName, l) < 0 {
-				// Insert the new module block here
-				updatedContent = append(updatedContent,
-					fmt.Sprintf("module \"%s\" {", o.PackageName),
-					fmt.Sprintf("  source            = \"%s\"", fmt.Sprintf(moduleSource, o.PackageName)),
-					fmt.Sprintf("  target_repository = \"%s/%s\"", targetRepository, o.PackageName), "}",
-					fmt.Sprintf(""), // this line will be added as a blank line after the new module added
-				)
-				moduleInserted = true
-			}
+	newTf := hclwrite.NewEmptyFile()
+	blocks := make(map[string]*hclwrite.Block)
+	for _, b := range f.Body().Blocks() {
+		switch b.Type() {
+		// extract modules to the map for sorting later
+		case "module":
+			blocks[b.Labels()[0]] = b
+		// write the block unchanged back to main.tf in the order it appears
+		default:
+			newTf.Body().AppendBlock(b)
+			newTf.Body().AppendNewline()
 		}
+	}
+	// add the new o.scaffold.PackageName module
+	blocks[newModule.Labels()[0]] = newModule
 
-		// Append the current line to the updated content
-		updatedContent = append(updatedContent, line)
+	// alphabetize the slice of block names
+	labels := []string{}
+	for k := range blocks {
+		labels = append(labels, k)
 	}
```

Comment:
```suggestion
	labels := maps.Keys(blocks)
```

https://pkg.go.dev/golang.org/x/exp/maps#Keys

Reply:
How did I not know about this? 🤯

Reply:
It's pretty new! And it's hidden away in /x/exp for now while they "stabilize the API" 🤷